### PR TITLE
Create preliminary routing documentation.

### PR DESCRIPTION
**Pull request by swizzlr**
@johanneserhardt @skreutzberger @gabrielPeart your feedback is welcomed and encouraged

### CODE REVIEW

File: docs/Routes.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Routing syntax
+
+## Simple matches
+
+Match path `/username` with any HTTP method.
+
+```swift
+route("/username")
```
Comment (skreutzberger) [inline docs/Routes.md]: What about getting rid of all slashes, be they at the end or beginning and instead always using a tuple for each part? In that way we could easier inject wildcards or variables?

Comment (gabrielPeart) [inline docs/Routes.md]: @skreutzberger nice point

Comment (swizzlr) [inline docs/Routes.md]: Agreed, though not mutually exclusive.

**Comment by swizzlr** [comment]
For context: 

Routing has not been written yet. I'm hoping I can steal either Frank, Kitura, Zewo or Vapor's matching code:

[Frank](https://github.com/nestproject/Frank)

```swift
// /users/{username}
get("users", *) { (request, username: String) in
  return "Hi \(username)"
}

// /users/{username}/followers
get("users", *, "followers") { (request, username: String) in
  return "\(username) has 5 followers"
}
```

**Comment by skreutzberger** [comment]
The way Frank is doing it looks good. Grouping we leave out for now I guess? I mean grouping all requests going to `/api/` for example?

**Comment by swizzlr** [comment]
[Kitura](https://github.com/IBM-Swift/Kitura/blob/master/Sources/Kitura/RouterHTTPVerbs_generated.swift): http://www.kitura.io/en/resources/tutorials/pathsyntax.html – tightly coupled with Templating engine of Kitura :(

[Zewo](https://github.com/Zewo/Zewo/blob/master/Modules/HTTPServer/Sources/HTTPServer/Router/TrieRouteMatcher.swift): Zewo has a really mature routing infrastructure using a `Trie` to implement once-only matching – something I would LOVE as a plugin to Titan later, but once-only matching (where it finds only one route and matches that) is out of scope for version 1. I think. It's also got this nice stuff here: https://github.com/Zewo/Zewo/blob/master/Modules/HTTPServer/Sources/HTTPServer/Router/Routes.swift

[Vapor](https://github.com/vapor/routing): Vapor has very mature matching as well, using a pseudo-Trie structure and some very clever generics. Unfortunately demonstrates complexity: includes WebSocket code handling.

**Pull request closed without merging**